### PR TITLE
Implement MultiLRScheduler

### LINKED ISSUE #50282: Add MultiScheduler
## 🚀 Feature
I would like to have a MultiScheduler that allows me to use multiple schedulers after each other, with the same optimizer.

## Motivation
It is not uncommon to have a learning rate schedule that looks like the following example:
- First have a burn in learning rate for 1000-2000 batches (eg. quadratic or exponential curve)
- Step down learning rate at steps 10.000 and 25.000.

You could implement all of this in a LambdaLR, but it would be a lot easier to combine a ``ExponentialLR`` and ``MultiStepLR``, running one after the other.

## Pitch
Implement a new LR-Scheduler, which I would call ``MultiScheduler`` (If anyone has a better name, I'm open to suggestions).
It receives a List of **N** schedulers, as well as a list **N-1** epoch numbers.
The scheduler can then automatically switch between different schedulers, depending on the `self.last_epoch` number.

I actually implemented such a scheduler in my own library and used it for quite a while now (ever since v0.4 IIRC), but I feel it deserves its place in the main PyTorch code base and am thus willing to make a PR that adds it!
However, unlike the current implementation, which takes a list of ``(epoch, scheduler)`` tuples, I think it would be cleaner take 2 separate lists.

## Alternatives
I do not think there are any viable alternatives to this problem, except for coding everything in a `LambdaLR` scheduler, which is far from ideal.

## Additional context
I don't know if there is any code in the PyTorch codebase that requires schedulers to have a ``self.optimizer`` variable, but as this scheduler combines multiple schedulers it has not.
However, I do use the base `_LRScheduler` class as a base and thus implement methods such as ``self.get_last_lr()``.

Let me know if you guys think it is a worthy addition to PyTorch, and I'll get started on my PR ! :construction_worker: 
Kind Regards,
0phoff

### PR DESCRIPTION
**Pull request by 0phoff**
Fixes #50282

This PR adds a MultiLRScheduler which allows you to combine multiple schedulers together.
Check the issue for more details.

- [X] MultiLRScheduler implementation
- [x] Unit tests 
- [ ] How to handle ReduceLROnPlateau

### CODE REVIEW

**Comment by 0phoff** [comment]
@mrshenli, @vincentqb Is there anything else that I forgot to do before this gets reviewed ? :innocent:
